### PR TITLE
docs(slide-toggle): migrate examples to MDC

### PR DESCRIPTION
**Pull request by zarend**
Migrate slide-toggle example to use mdc-based components.

### CODE REVIEW

File: src/dev-app/slide-toggle/slide-toggle-demo.html [hunk 1]
```diff
@@ -1,29 +1,8 @@
...
-    <p>
-      <button mat-raised-button type="submit">Submit Form</button>
-    </p>
-
-  </form>
-
-</div>
+<h3>Slide Toggle Overview</h3>
+<slide-toggle-overview-example></slide-toggle-overview-example>
+<h3>Configurable Slide Toggle</h3>
```
Comment (crisbeto) [inline src/dev-app/slide-toggle/slide-toggle-demo.html:3]: Can we keep the old inline demo? I use the dev app for debugging and these demos that are based on the docs examples tend to be annoying to debug. We already have the `/examples` page where all the examples can be seen.

Comment (zarend) [reply src/dev-app/slide-toggle/slide-toggle-demo.html:3]: Sure, I'll do that then request review again.

**Comment by zarend** [comment]
- brought back the inline demo
 - fixed component harness example
 This PR is ready for review again.

**Review by crisbeto — APPROVED** [review]
LGTM

**Pull request merged**